### ISSUE TITLE
Filename incorrectly parsed from Content-Disposition header

### ISSUE THREAD

**Issue by kylefox** [pqina/filepond#64]
First of all — awesome plugin 👍 

We use Amazon S3 for file storage, and the `Content-Disposition` header includes both the `filename` and [`filename*`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition#Directives) directives:

```
Content-Disposition: inline; filename="airplane.png"; filename*=UTF-8''airplane.png
```

Which displays like so in FilePond:

![image](https://user-images.githubusercontent.com/37838/41562725-b4841392-730a-11e8-9f00-2eb4302f8d9a.png)

A potential fix could be to explicitly search for the `filename=""` directive using a regular expression:

```javascript
const getFilenameFromHeaders = headers => {
  const rows = headers.split('\n');
  for (const header of rows) {
    const match = header.match(/filename="(.*)"/)
    if (!match) {
      continue;
    }
    return match[1].replace(/["']+/g, '');
  }
  return null;
};
```

Happy to submit a pull request if you'd prefer that. Thanks!

**Comment by rikschennink** [comment]
Will close for now. If not resolved, I'll re-open.

**Comment by kylefox** [comment]
Works perfectly 👍Thanks for the quick response!

**Comment by dnknitro** [comment]
I am using v4.4.0:
        "filepond": "^4.4.0",
        "filepond-plugin-file-metadata": "^1.0.6",
        "filepond-plugin-image-preview": "^4.0.8",
and it is still an issue. [See screenshot](https://i.imgur.com/gqQiaK7.png)

**Comment by petrama** [comment]
> I am using v4.4.0:
> "filepond": "^4.4.0",
> "filepond-plugin-file-metadata": "^1.0.6",
> "filepond-plugin-image-preview": "^4.0.8",
> and it is still an issue. [See screenshot](https://i.imgur.com/gqQiaK7.png)

Same in v4.3.9

**Comment by rikschennink** [comment]
Please paste the content disposition header of the response so I can try to reproduce the issue.

**Comment by dnknitro** [comment]
Content-Disposition: attachment; filename=Invoker.jpg; filename*=UTF-8''Invoker.jpg

**Comment by rikschennink** [comment]
@dnknitro @petrama I've confirmed the issue and will probably roll out a fix next week.

**Comment by rikschennink** [comment]
Fixed in `4.4.1`

**Comment by dnknitro** [comment]
I have just tried the fix and filename with spaces (or probably any other url encoded characters are still displayed incorrectly.
Header: `Content-Disposition: attachment; filename="wp SecretaryBird 1920x1200.jpg"; filename*=UTF-8''wp%20SecretaryBird%201920x1200.jpg`
Filename displayed: `wp%20SecretaryBird%201920x1200.jpg`

**Comment by rikschennink** [comment]
@dnknitro I've added this as a test case and now it should be fixed in `4.4.2`

**Comment by dnknitro** [comment]
Works now, thank you.
